src/Symfony/Component/PropertyInfo/Extractor/PhpStanExtractor.php

```diff
@@ -86,11 +90,12 @@ public function getTypes(string $class, string $property, array $context = []):
     {
         /** @var PhpDocNode|null $docNode */
         [$docNode, $source, $prefix, $declaringClass] = $this->getDocBlock($class, $property);
-        $nameScope = $this->nameScopeFactory->create($class, $declaringClass);
         if (null === $docNode) {
             return null;
         }
 
+        $nameScope = $this->contexts[$class.$declaringClass] ??= $this->nameScopeFactory->create($class, $declaringClass);
```

Comment:
```suggestion
        $nameScope = $this->contexts[$class.'/'.$declaringClass] ??= $this->nameScopeFactory->create($class, $declaringClass);
```

Reply:
would it make sense to move this inside the loop, right before the nested foreach?:

```php
$nameScope ??= $this->contexts[$class.'/'.$declaringClass] ??= $this->nameScopeFactory->create($class, $declaringClass);
```

Reply:
> would it make sense to move this inside the loop, right before the nested foreach?:

You mean here? 

https://github.com/symfony/symfony/blob/de1498080d1b34a289e887eddceeada5e1b30087/src/Symfony/Component/PropertyInfo/Extractor/PhpStanExtractor.php#L127

Sounds reasonable to me. It may make sense to skip creating one unless really needed (and since my change is all about re-using a created one, it hardly influences performance).